### PR TITLE
CStats finished

### PR DESCRIPTION
**Pull request by Sergeanur**

### CODE REVIEW

**Comment by ShFil119** [comment]
Needs rebasing.

File: src/core/Stats.cpp [hunk 1]
```diff
@@ -200,6 +200,229 @@ void CStats::SetTotalNumberMissions(int32 total)
...
+
+	rating = FiresExtinguished + 10 * HighestLevelAmbulanceMission + CriminalsCaught + LivesSavedWithAmbulance
+		+ 30 * HelisDestroyed + TotalLegitimateKills - 3 * TimesArrested - 3 * TimesDied
+		+ CWorld::Players[CWorld::PlayerInFocus].m_nMoney / 5000;
+	if (rating <= 0) rating = 0;
+
+	if (InstantHitsFiredByPlayer > 100)
+		rating += (double)CStats::InstantHitsHitByPlayer / (double)CStats::InstantHitsFiredByPlayer * 500.0f;
+	if (TotalProgressInGame)
+		rating += (double)CStats::ProgressMade / (double)CStats::TotalProgressInGame * 1000.0f;
```
Comment (ShFil119) [inline src/core/Stats.cpp:236]: mobile is using floats here. not sure if we need doubles

**Comment by erorcun** [comment]
lgtm

File: src/core/Stats.cpp (hunk 1 above)
Comment (aap) [reply src/core/Stats.cpp:236]: I assume this is a case of missing f-suffix on the constants. everything gets promoted to double in that case. We should just make it float.

**Pull request merged**